langserve/server.py

```diff
@@ -456,204 +457,163 @@ def _json_encode_response(model: BaseModel) -> JSONResponse:
     return JSONResponse(content=obj)
 
 
-# PUBLIC API
-
-
-def add_routes(
-    app: Union[FastAPI, APIRouter],
-    runnable: Runnable,
-    *,
-    path: str = "",
-    input_type: Union[Type, Literal["auto"], BaseModel] = "auto",
-    output_type: Union[Type, Literal["auto"], BaseModel] = "auto",
-    config_keys: Sequence[str] = ("configurable",),
-    include_callback_events: bool = False,
-    enable_feedback_endpoint: bool = False,
-    per_req_config_modifier: Optional[PerRequestConfigModifier] = None,
-) -> None:
-    """Register the routes on the given FastAPI app or APIRouter.
+try:
+    from sse_starlette import EventSourceResponse
+except ImportError:
+    EventSourceResponse = Any
 
 
-    The following routes are added per runnable under the specified `path`:
+class _APIHandler:
```

Comment:
does this really need to be private? can we move into its own file?

Reply:
I'd prefer private until it's obvious that this is useful to users and we decide that the API is correct? I feel like we're likely to change this again soonish